pylib/anki/cards.py

```diff
@@ -216,6 +217,15 @@ def set_user_flag(self, flag: int) -> None:
             raise Exception("invalid flag")
         self.flags = (self.flags & ~0b111) | flag
 
+    def review_hash(self) -> str:
+        """Sha1 of this card's id and repetition count."""
+        hash = hashlib.sha1()
+        # CardId is an i64 in Rust
+        hash.update(self.id.to_bytes(8, "little", signed=True))
+        # reps are a u32 in Rust
+        hash.update(self.reps.to_bytes(4, "little"))
+        return hash.hexdigest()
```

Comment:
Could we move this into Rust, so that we don't need to reimplement it/keep it up to date across the various clients? Maybe something like col.backend.get_scheduling_context(c.id, c.reps)?

A full sha1 hex digest is also probably excessive, as FSRS tucks the seed into the custom card data, and this is going to add a fair bit to the storage of each row. How about a u64, like we use for fuzz_seed? JS won't be able to represent values exactly if it's larger than 52 bits, but since it's a random number, perhaps that doesn't matter?

Reply:
Makes sense.
However, it seemed just adding id and reps as for the fuzz seed is too prone to collisions so I've added a little tweak.